### PR TITLE
 Initial version of the end-to-end tests.

### PR DESCRIPTION
**Pull request by jvanz**

### CODE REVIEW

File: .github/workflows/e2e-tests.yml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+      policy-server-container-image-artifact:
+        description: "Define the artifact containing the policy server container image"
+        type: string
+        required: false
+        default: ""
+      end-to-end-tests-version:
+        description: "The end-to-end tests version tag to be run"
+        type: string
+        required: false
+        default: "v0.1.0"
```
Comment (jvanz) [inline .github/workflows/e2e-tests.yml:40]: I'll create the tag once the PR is merged

**Comment by jvanz** [comment]
Related to https://github.com/kubewarden/kubewarden-controller/issues/125

**Review by viccuad — APPROVED** [review]
Approving.
As said also in https://github.com/kubewarden/setup-kubewarden-cluster-action/pull/1#pullrequestreview-821287066,
I'm totally ok with this implementation if a majority agrees, but I still have the same concerns listed in kubewarden/kubewarden-controller#127 (review). From previous CI work, I think it's a good idea to future-proof by allowing to partially run them against any cluster and stack, outside of GHA.
In addition, we have a bunch of shellscript there (and we may have other things in the near future). I would prefer if those snippets where in their files, and we had linting for them (shellcheck, etc).

**Comment by viccuad** [comment]
I was hoping for other reviews ("if a majority aggres"), before this got merged to be fair!

**Pull request merged**